### PR TITLE
NMS-10345: Added feature to report vSphere alarms in VmwareMonitor

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-10345

### CODE REVIEW

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 1]
```diff
@@ -268,6 +268,7 @@
          <parameter key="retry" value="2"/>
          <parameter key="timeout" value="3000"/>
          <parameter key="ignoreStandBy" value="false"/>
+         <parameter key="reportAlarms" value=""/>
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml]: Is it required to force a merge for updating users with an empty new introduced parameter?

Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml]: no, empty string is the default

Comment (indigo423) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml]: You could avoid touching the config-tester, by making reportAlarms="" when the whole parameter is missing and make it backward compatible.

Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml]: I think I understood you correctly, the default value "" is also applied in code even the reportAlarms entry does not exist. Of course we can also remove the line here if you feel better :-)

File: integrations/opennms-vmware/src/main/java/org/opennms/netmgt/poller/monitors/VmwareMonitor.java [hunk 2]
```diff
@@ -83,6 +98,16 @@
...
         final boolean ignoreStandBy = getKeyedBoolean(parameters, "ignoreStandBy", false);
+
+        final List<String> severitiesToReport =
+                Splitter.on(",")
+                    .trimResults()
+                    .omitEmptyStrings()
+                    .splitToList(getKeyedString(parameters, "reportAlarms", ""))
+                    .stream()
+                    .filter(e -> VALID_VSPHERE_ALARM_STATES.contains(e))
+                    .collect(Collectors.toList());
```
Comment (christianpape) [inline integrations/opennms-vmware/src/main/java/org/opennms/netmgt/poller/monitors/VmwareMonitor.java]: Here it is: getKeyedString(parameters, "reportAlarms", "") defaults to an empty string

**Comment by indigo423** [comment]
I have tears in my eyes for this PR which includes documentation :)
@christianpape🥇

**Pull request merged**